### ISSUE TITLE
Fresh install of ROS2 Galactic fails to load _rclpy_pybind11.cpython-38-x86_64-linux-gnu.so due to wrong Python version in release artifacts.

### ISSUE THREAD

**Issue by hartmannathan** [ros2/ros2#1167]
## Bug report

tl;dr: The Galactic 2021-07-16 release artifact was generated on a machine which targeted the wrong Python version.

NB: This is not a duplicate of #1164 but might be related.

**Required Info:**

- Operating System:
  - Ubuntu 20.04 x86_64
- Installation type:
  - Fresh install of Ubuntu OS
  - Fresh install of ROS Galactic following instructions at: https://docs.ros.org/en/galactic/Installation/Ubuntu-Install-Binary.html
- Version or commit hash:
  - This release artifact: https://github.com/ros2/ros2/releases/download/release-galactic-20210716/ros2-galactic-20210616-linux-focal-amd64.tar.bz2
- DDS implementation:
  - rti-connext-dds-5.3.1
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue

Fresh OS install.

Followed instructions at: https://docs.ros.org/en/galactic/Installation/Ubuntu-Install-Binary.html

Attempted to run the basic examples:

Terminal 1:

```
. ~/ros2_galactic/ros2-linux/setup.bash
ros2 run demo_nodes_cpp talker
```

Terminal 2:

```
. ~/ros2_galactic/ros2-linux/setup.bash
ros2 run demo_nodes_py listener
```

#### Expected behavior

Expected the example C and Python programs to run and show messages sent/received.

#### Actual behavior

The Python program (demo_nodes_py) will not run because of mismatched Python versions.

Ubuntu 20.04 comes with Python 3.8.

The ROS download named above targets Python 3.6 (i.e., it contains the directory `ros-linux/lib/python3.6` instead of `ros-linux/lib/python3.8`). That is incorrect for the targeted OS version.

#### Additional information

Here is the traceback:

```
user@robot:~$ ros2 run demo_nodes_py listener
Traceback (most recent call last):
  File "/home/user/ros2_galactic/ros2-linux/lib/demo_nodes_py/listener", line 33, in <module>
    sys.exit(load_entry_point('demo-nodes-py==0.14.3', 'console_scripts', 'listener')())
  File "/home/user/ros2_galactic/ros2-linux/lib/demo_nodes_py/listener", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/user/ros2_galactic/ros2-linux/lib/python3.6/site-packages/demo_nodes_py/topics/listener.py", line 16, in <module>
    from rclpy.node import Node
  File "/home/user/ros2_galactic/ros2-linux/lib/python3.6/site-packages/rclpy/node.py", line 43, in <module>
    from rclpy.client import Client
  File "/home/user/ros2_galactic/ros2-linux/lib/python3.6/site-packages/rclpy/client.py", line 22, in <module>
    from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
  File "/home/user/ros2_galactic/ros2-linux/lib/python3.6/site-packages/rclpy/impl/implementation_singleton.py", line 32, in <module>
    rclpy_implementation = import_c_library('._rclpy_pybind11', package)
  File "/home/user/ros2_galactic/ros2-linux/lib/python3.6/site-packages/rpyutils/import_c_library.py", line 39, in import_c_library
    return importlib.import_module(name, package=package)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named 'rclpy._rclpy_pybind11'
The C extension '/home/user/ros2_galactic/ros2-linux/lib/python3.6/site-packages/_rclpy_pybind11.cpython-38-x86_64-linux-gnu.so' isn't present on the system. Please refer to 'https://index.ros.org/doc/ros2/Troubleshooting/Installation-Troubleshooting/#import-failing-without-library-present-on-the-system' for possible solutions
```

Note: Installing via Debian packages provides the correct directory, `/opt/ros/galactic/lib/python3.8`, but installing via the above release artifact does not work for reasons stated above.

## Feature request

N/A

#### Feature description

N/A

#### Implementation considerations

N/A

**Comment by cottsay** [comment]
Thanks for bringing this to our attention, @hartmannathan. It looks like the Focal archive got swapped with the CentOS archive. I've renamed them appropriately on the GitHub release - please try downloading the archive again and re-open this issue if the problem persists.
